tools/targets/STM32_gen_PeripheralPins.py

```diff
@@ -1343,27 +1477,47 @@ def parse_BoardFile(fileName):
         TargetName += "DISCO_"
     elif "Evaluation" in board_file_name:
         TargetName += "EVAL_"
-    m = re.search(r'STM32([\w]{1,2}[\dR]{3}[\w]{0,2})[\w]*_Board', board_file_name)
+    m = re.search(r'STM32([MFLGWH][\w]*)_Board', board_file_name)
     if m:
         TargetName += "%s" % m.group(1)
         # specific case
-        if "-P" in args.target:
+        if "-P" in board_file_name:
             TargetName += "_P"
-
-        if "H743ZI2" in board_file_name:
-            TargetName += "2"
+        elif "-Q" in board_file_name:
+            TargetName += "_Q"
 
         if TargetName == "DISCO_L072C":
             TargetName += "Z_LRWAN1"
         elif TargetName == "DISCO_L475V":
             TargetName += "G_IOT01A"
         elif TargetName == "DISCO_G071RBT":
             TargetName = "DISCO_G071RB"
-        elif TargetName == "DISCO_H747XI":
+        elif TargetName == "DISCO_L4R9A":
```

Comment:
Should we not define a `target_name_map` as a dictionary mapping these target names? Then we wouldn't need all of these `if` statements.